pkg/ceph/osd/device.go

```diff
@@ -674,9 +723,29 @@ func getBluestorePartitionPaths(config *osdConfig) (string, string, string, erro
 		return "", "", "", fmt.Errorf("failed to find block partition for osd %d", config.id)
 	}
 
-	return path.Join(diskByPartUUID, walPartition.PartitionUUID),
-		path.Join(diskByPartUUID, dbPartition.PartitionUUID),
-		path.Join(diskByPartUUID, blockPartition.PartitionUUID),
+	return filepath.Join(diskByPartUUID, walPartition.PartitionUUID),
+		filepath.Join(diskByPartUUID, dbPartition.PartitionUUID),
+		filepath.Join(diskByPartUUID, blockPartition.PartitionUUID),
+		nil
+
+}
+
+func getBluestoreDirPaths(config *osdConfig) (string, string, string, error) {
+	if !isBluestoreDir(config) {
+		return "", "", "", fmt.Errorf("must be bluestore dir to get bluestore dir paths: %+v", config)
+	}
+
+	return filepath.Join(config.rootPath, bluestoreDirWalName),
+		filepath.Join(config.rootPath, bluestoreDirDBName),
+		filepath.Join(config.rootPath, bluestoreDirBlockName),
 		nil
+}
+
+func getSizeForPath(osdDataPath string) (uint64, error) {
+	s := syscall.Statfs_t{}
+	if err := syscall.Statfs(osdDataPath, &s); err != nil {
```

Comment:
should this go in the `sys` package?

Reply:
makes sense to put it in the `sys` package.

Reply:
ah, i ran into a build error for non linux platforms, because `syscall.Statfs` doesn't exist there.  i'm going to leave it here since only the OSDs are using it.